src/Numerical/RootFinding/QuadraticEquation.cpp

```diff
@@ -0,0 +1,24 @@
+
+#include "Numerical/RootFinding/QuadraticEquation.hpp"
+
+#include <gsl/gsl_poly.h>
+
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/ExpectsAndEnsures.hpp"
+
+double positive_root(const double a, const double b, const double c) {
+  double x0;
+  double x1;
+  gsl_poly_solve_quadratic(a, b, c, &x0, &x1);
+  Ensures(x0 <= 0.0 and x1 >= 0.0);
+  return x1;
+}
+
+std::array<double, 2> real_roots(const double a, const double b,
+                                 const double c) {
+  double x0;
+  double x1;
+  int numrealroots = gsl_poly_solve_quadratic(a, b, c, &x0, &x1);
+  ASSERT(numrealroots == 2, "Assumes that there are two real roots");
+  return {{x0, x1}};
```

Comment:
See above comment re: variable assignment. The `ASSERT` makes this code safe in some builds only.